hilti/toolchain/src/compiler/parser/parser.yy

```diff
@@ -789,7 +793,7 @@ expr_f        : ctor                             { $$ = hilti::expression::Ctor(
                                                  { $$ = hilti::expression::ListComprehension(std::move($6), std::move($2), std::move($4), {},  __loc__); }
               | '[' expr FOR local_id IN expr IF expr ']'
                                                  { $$ = hilti::expression::ListComprehension(std::move($6), std::move($2), std::move($4), std::move($8),  __loc__); }
-              | expr_g
+              | expr_g                           { $$ = std::move($1); }
```

Comment:
🤦 , of course.

This patch looks like what you want. Let's merge it after cleanup and pulling back in the unordered collection for the location cache (a benchmark shows that that's worthwhile).

Reply:
Yep, will do, reverted it only for debugging.